src/render/src/pso/buffer.rs

```diff
@@ -104,21 +104,50 @@ pub struct ConstantBuffer<T: Structure<shade::ConstFormat>>(
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
 pub struct RawConstantBuffer(Option<(Usage, ConstantBufferSlot)>);
 
-/// Global (uniform) constant component. Describes a free-standing value passed into
-/// the shader, which is not enclosed into any constant buffer. Deprecated in DX10 and higher.
+/// Global (uniform) constant component. Describes a free-standing value passed
+/// into the shader, which is not enclosed into any constant buffer.
 ///
 /// - init: `&str` = name of the constant
 /// - data: `T` = value
 #[derive(Derivative)]
 #[derivative(Clone, Debug, Eq, Hash, PartialEq)]
 pub struct Global<T: ToUniform>(
-    Option<shade::Location>,
+    RawGlobal,
     #[derivative(Hash = "ignore", PartialEq = "ignore")]
     PhantomData<T>
 );
 
-fn match_attribute(_: &shade::AttributeVar, _: Format) -> bool {
-    true //TODO
+/// Raw global (uniform) constant component. Describes a free-standing value
+/// passed into the shader, which is not enclosed in any constant buffer.
+///
+/// - init: `&str` = name of the constant
+/// - data: `UniformValue` = value
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct RawGlobal(Option<shade::Location>);
+
+fn match_attribute(attr: &shade::AttributeVar, fmt: Format) -> bool {
```

Comment:
1. let's leave this for the follow-ups
  2. nope, those can't be attributes
  3. yes